dotnetv3/StepFunctions/Actions/StepFunctionsWrapper.cs

```diff
@@ -0,0 +1,283 @@
+    /// <returns>The response from the Step Functions activity.</returns>
+    public async Task<GetActivityTaskResponse> GetActivityTaskAsync(string activityArn, string workerName)
+    {
+        var response = await _amazonStepFunctions.GetActivityTaskAsync(new GetActivityTaskRequest
+        { ActivityArn = activityArn, WorkerName = workerName });
+        return response;
+    }
+
+    // snippet-end:[StepFunctions.dotnetv3.GetActivityTask]
+
+    // snippet-start:[StepFunctions.dotnetv3.ListActivities]
+    /// <summary>
+    /// List the Step Functions activities for the current account.
+    /// </summary>
+    /// <returns>A list of ActivityListItems.</returns>
+    public async Task<List<ActivityListItem>> ListActivities()
+    {
+        Console.WriteLine("Welcome to Amazon Step Functions. Let's list your Step Functions actions:");
+        var request = new ListActivitiesRequest { MaxResults = 10 };
+        var activities = new List<ActivityListItem>();
+
+        do
+        {
+            var response = await _amazonStepFunctions.ListActivitiesAsync(request);
+
+            if (response.NextToken is not null)
+            {
+                request.NextToken = response.NextToken;
+            }
+
+            activities.AddRange(response.Activities);
+        }
+        while (request.NextToken is not null);
+
+        return activities;
+    }
+
+    // snippet-end:[StepFunctions.dotnetv3.ListActivities]
+
+    // snippet-start:[StepFunctions.dotnetv3.ListExecutions]
+    /// <summary>
+    /// Retrieve information about executions of a Step Functions
+    /// state machine.
+    /// </summary>
+    /// <param name="stateMachineArn">The Amazon Resource Name (ARN) of the
+    /// Step Functions state machine.</param>
+    /// <returns>A list of ExecutionListItem objects.</returns>
+    public async Task<List<ExecutionListItem>> ListExecutionsAsync(string stateMachineArn)
+    {
+        var executions = new List<ExecutionListItem>();
+        var response = new ListExecutionsResponse();
+        var request = new ListExecutionsRequest { StateMachineArn = stateMachineArn };
+
+        do
+        {
+            response = await _amazonStepFunctions.ListExecutionsAsync(new ListExecutionsRequest
+            { StateMachineArn = stateMachineArn });
+            executions.AddRange(response.Executions);
+            if (response.NextToken is not null)
+            {
+                request.NextToken = response.NextToken;
+            }
+        } while (response.NextToken is not null);
+
+        return executions;
+    }
+
+    // snippet-end:[StepFunctions.dotnetv3.ListExecutions]
+
+    // snippet-start:[StepFunctions.dotnetv3.ListStateMachines]
+    /// <summary>
+    /// Retrieve a list of Step Funcsions state machines.
+    /// </summary>
+    /// <returns>A list of StateMachineListItem objects.</returns>
+    public async Task<List<StateMachineListItem>> ListStateMachinesAsync()
```

Comment:
Could not find this method in the scenario or the tests.

Reply:
It's the same as the Hello Step Functions example so it's not in the scenario. Added the test.